### PR TITLE
Adds Facilities Connect Update Status Endpoint

### PR DESCRIPTION
**Pull request by lspdrz**
## Proposed changes

Allows DOC staff to edit the statuses of connections at their facility

Closes Issue #15 

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: src/authorization/guards/staff-authorization.guard.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export class StaffGuard implements CanActivate {
   async canActivate(context: ExecutionContext): Promise<boolean> {
     const request = context.switchToHttp().getRequest();
     const { user } = request;
-    if (!user.staffPositions.length) return false;
+    if (!user.staffPositions || !user.staffPositions.length) return false;
```
Comment (gabrielsaruhashi) [inline src/authorization/guards/staff-authorization.guard.ts:9]: do we need the !user.staffPositions chekc here? 
I think user.staffPositions returns an empty array [] if there are no positions

File: src/facilities/facilities.service.ts [hunk 2]
```diff
@@ -61,6 +66,38 @@ export default class FacilitiesService {
...
+    );
+    if (connection.facilityId !== facilityId) {
+      throw new HttpException(
+        "Connection doesn't belong to Facility",
+        HttpStatus.FORBIDDEN,
+      );
+    }
+    const ability = this.caslAbilityFactory.createForStaff(staffMember);
+    const isAuthorized =
+      ability.can(Action.Update, connection, 'status') &&
```
[the remark this answers is not in the document]
Comment (gabrielsaruhashi) [reply src/facilities/facilities.service.ts:86]: Okie! I'll try to refactor as I work on CASL

File: src/authorization/casl/staff/staff-ability.casl.ts [hunk 3]
```diff
@@ -22,13 +30,23 @@ const getStaffAbility = (staff: Staff): StaffAbility => {
...
-  //Inmates
+  // Inmates
   if (staff.role === StaffRole.ADMIN) {
     can(Action.Create, Inmate, {
       facility: staff.facility,
     });
   }
 
+  // Connections
+  if (
```
Comment (gabrielsaruhashi) [inline src/authorization/casl/staff/staff-ability.casl.ts:41]: should move the staff.isActive to the guard then?

**Review by gabrielsaruhashi — APPROVED** [review]
loooks great! one final q/comment

File: src/authorization/casl/staff/staff-ability.casl.ts (hunk 3 above)
Comment (lspdrz) [reply src/authorization/casl/staff/staff-ability.casl.ts:41]: Let's have a discussion about this later, but it makes sense to me conceptually

**Pull request merged**